plugins/pyembed/pyembed.cpp

```diff
@@ -84,6 +85,25 @@ class OwnedPyObject
     inline PyObject **ref()                 { return &ptr; }
 };
 
+template <class X>
+class OwnedPyX
+{
+    X *ptr;
+public:
+    inline OwnedPyX<X>() : ptr(NULL) {}
+    inline OwnedPyX<X>(X *_ptr) : ptr(_ptr) {}
+    inline ~OwnedPyX<X>()            { if (ptr) Py_DECREF(ptr); }
+    inline X * get() const           { return ptr; }
+    inline X * getClear()            { PyObject *ret = ptr; ptr = NULL; return ret; }
+    inline X * operator -> () const  { return ptr; }
+    inline operator X *() const      { return ptr; }
+    inline void clear()              { if (ptr) Py_DECREF(ptr); ptr = NULL; }
+    inline void setown(X *_ptr)      { clear(); ptr = _ptr; }
+    inline void set(X *_ptr)         { clear(); ptr = _ptr; if (ptr) Py_INCREF(ptr);}
```

Comment:
Minor: If you do value.set(value) it will unlink the value before linking the new value.  Unlikely to happen in practice for this class.